### PR TITLE
Add unit tests for ProgressTableDetailCell

### PR DESCRIPTION
**Pull request by maureensturgeon**
Add unit tests for ProgressTableDetailCell

## Links

- [spec]()
- [jira]()

## Testing story

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/test/unit/templates/sectionProgress/progressTables/ProgressTableDetailCellTest.jsx
Comment (cforkish) [inline apps/test/unit/templates/sectionProgress/progressTables/ProgressTableDetailCellTest.jsx:132]: could you add a test to make sure the unplugged bubble properly resizes for a different locale? i only ask because the unplugged bubble used to have a fixed size, but i updated it to (hopefully) size perfectly to the localized text.

File: apps/test/unit/templates/sectionProgress/progressTables/ProgressTableDetailCellTest.jsx
Comment (cforkish) [inline apps/test/unit/templates/sectionProgress/progressTables/ProgressTableDetailCellTest.jsx]: if it's not too much work, could you import the various constants that are used to calculate these widths rather than testing with these magic numbers?

**Review by cforkish — CHANGES_REQUESTED** [review]
a couple minor comments. feel free to push back if either of them turn out to be an overly-heavy lift!

also, per our previous discussion, i want to make sure that in the end we have as little duplicated test data as possible. i think it would be fine to merge these tests into staging as-is though, even if you plan to refactor the test data in a subsequent PR, so for purposes of this PR i'm fine with leaving it how it is.

File: apps/test/unit/templates/sectionProgress/progressTables/ProgressTableDetailCellTest.jsx
Comment (maureensturgeon) [reply apps/test/unit/templates/sectionProgress/progressTables/ProgressTableDetailCellTest.jsx]: The only test where this is tricky are for the unplugged levels, the width is calculated by https://github.com/code-dot-org/code-dot-org/blob/cforkish/LP-1604-refactor-progress-tables/apps/src/templates/sectionProgress/progressTables/ProgressTableDetailCell.jsx#L168. Do you think I should export this function and use it in the test?

I updated the other 2.

Comment (cforkish) [reply apps/test/unit/templates/sectionProgress/progressTables/ProgressTableDetailCellTest.jsx]: why don't you go ahead and use our `unitTestExports` convention to export that function. however, since there's no constant to compare to, just put a comment saying the 250 is the observed render width for the english localization.

then in a separate test you can choose a language with a noticeably different localization length, and expect that it's wider or narrower than the english localization.

**Review by jamescodeorg — COMMENTED** [review]
I'm going to defer to Charlie on this review.

File: apps/test/unit/templates/sectionProgress/progressTables/ProgressTableDetailCellTest.jsx
Comment (maureensturgeon) [reply apps/test/unit/templates/sectionProgress/progressTables/ProgressTableDetailCellTest.jsx]: I exported the `getUnpluggedWidth` function and tested it independently and then was able to use it to determine the expected width in the unit test.

File: apps/test/unit/templates/sectionProgress/progressTables/ProgressTableDetailCellTest.jsx
Comment (cforkish) [inline apps/test/unit/templates/sectionProgress/progressTables/ProgressTableDetailCellTest.jsx:145]: 🙌🎉

File: apps/test/unit/templates/sectionProgress/progressTables/ProgressTableDetailCellTest.jsx
Comment (cforkish) [inline apps/test/unit/templates/sectionProgress/progressTables/ProgressTableDetailCellTest.jsx]: i think this is a copy/paste typo. reading "there are no unplugged levels.... unpluggedLevelCount = 2" kinda tripped me up.

File: apps/test/unit/templates/sectionProgress/progressTables/ProgressTableDetailCellTest.jsx
Comment (cforkish) [inline apps/test/unit/templates/sectionProgress/progressTables/ProgressTableDetailCellTest.jsx]: now that i've fixed these in 1604, let's not merge any more instances in. `to.be.true` and `to.be.false` please!

**Review by cforkish — APPROVED** [review]
nice work with the unplugged tests! i found a few little nits, but approving so you can go ahead and merge once you update

**Pull request merged**
